custom_components/samsungtv_tizen/media_player.py

```diff
@@ -309,11 +350,21 @@ def _gen_installed_app_list(self):
             except Exception:
                 pass
 
-        self._app_list = clean_app_list
+        #self._app_list = clean_app_list
+        self._app_list_ST = self._app_list = clean_app_list
         _LOGGER.debug("Gen installed app_list %s", clean_app_list)
 
     def _get_source(self):
+        """Return the current input source."""
         if self._state != STATE_OFF:
+
+            call_time = datetime.now()
```

Comment:
I don't think there is any reason to throttle the `_get_source()` method, it does no requests at all, the method itself takes very few milliseconds to complete. The only method that we should be careful to not overuse is `_get_running_app()`, which can do multiple HTTP requests.

Reply:
I throttle this method for a different reason. When I select a new source (executing a new command) I think is better to leave 5 second for the component to refresh the real selected source and I temporary force the source to the new value (line 721). In this mode the source selected reflect immediatly the choice and do not create confusion.

Reply:
I see, we should put that in a comment next to this code, it can be easily confusing otherwise and might be removed in time if it isn't explained properly.

Reply:
Comment is always good, that's my fault :)

Reply:
Added some comment about this in the code